### PR TITLE
Added methods to create categories with json arguments

### PR DESCRIPTION
**Pull request by scand1n**

### CODE REVIEW

**Comment by scand1n** [comment]
update build.yml

File: CloudFabric.EAV.Service/EAVService.cs [hunk 1]
```diff
@@ -976,10 +976,321 @@ await _attributeConfigurationRepository
...
+
+        if (validationErrors != null)
+        {
+            return (null, validationErrors);
+        }
+
+        return (SerializeEntityInstanceToJsonMultiLanguage(_mapper.Map<EntityInstance>(createdCategory)), null);
+    }
+
     public async Task<(CategoryViewModel, ProblemDetails)> CreateCategoryInstanceAsync(
```
Comment (scand1n) [inline CloudFabric.EAV.Service/EAVService.cs:1281]: Remove "Async" from method naming

File: CloudFabric.EAV.Service/EAVService.cs [hunk 2]
```diff
@@ -976,10 +976,321 @@ await _attributeConfigurationRepository
...
+    /// <param name="requestDeserializedCallback">This function will be called after deserializing the request from json
+    /// to CategoryInstanceCreateRequest and allows adding additional validation or any other pre-processing logic.
+    /// </param>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    public Task<(JsonDocument?, ProblemDetails?)> CreateCategoryInstance(
+    string categoryJsonString,
+    Func<CategoryInstanceCreateRequest, Task<CategoryInstanceCreateRequest>>? requestDeserializedCallback = null,
+    CancellationToken cancellationToken = default
+)
```
Comment (ustims) [inline CloudFabric.EAV.Service/EAVService.cs:1015]: Bad formatting

File: CloudFabric.EAV.Service/EAVService.cs [hunk 3]
```diff
@@ -976,10 +976,321 @@ await _attributeConfigurationRepository
...
+    ///
+    /// ```
+    /// {
+    ///     "name": "Main Category",
+    ///     "desprition": "Main Category description"
+    /// }
+    /// Where "name" and "description" are attributes machine names.
+    /// Note that this overload accepts "entityConfigurationId" and "tenantId" via method arguments,
+    /// so they should not be in json.
+    /// ```
```
Comment (ustims) [inline CloudFabric.EAV.Service/EAVService.cs:1040]: \`\`\` should be moved above the comment to only include json in \`\`\`... \`\`\` section

File: CloudFabric.EAV.Service/EAVService.cs [hunk 4]
```diff
@@ -976,10 +976,321 @@ await _attributeConfigurationRepository
...
+    ///
+    /// Where "name" and "description" are attributes machine names,
+    /// "entityConfigurationId" - obviously the id of entity configuration which has all category attributes,
+    /// "categoryTreeId" - guid of category tree, which represents separated hirerarchy with relations between categories
+    /// "parentId" - id guid of category from which new branch of hierarchy will be built.
+    /// Can be null if placed at the root of category tree.
+    /// "tenantId" - tenant id guid. A guid which uniquely identifies and isolates the data. For single tenant
+    /// application this should be one hardcoded guid for whole app.
+    ///
+    /// ```
```
Comment (ustims) [inline CloudFabric.EAV.Service/EAVService.cs:1100]: \`\`\` should be moved above the comment to only include json in \`\`\`... \`\`\` section

File: CloudFabric.EAV.Service/EAVService.cs [hunk 5]
```diff
@@ -976,10 +976,321 @@ await _attributeConfigurationRepository
...
+    /// {
+    ///     "name": "Main Category",
+    ///     "desprition": "Main Category description"
+    /// }
+    /// Where "name" and "description" are attributes machine names.
+    /// Note that this overload accepts "entityConfigurationId" and "tenantId" via method arguments,
+    /// so they should not be in json.
+    /// ```
```
Comment (ustims) [inline CloudFabric.EAV.Service/EAVService.cs:1215]: \`\`\` should be moved above the comment to only include json in \`\`\`... \`\`\` section

File: CloudFabric.EAV.Tests/JsonSerializationTests.cs [hunk 6]
```diff
@@ -254,4 +255,71 @@ public async Task TestCreateInstanceSingleLangAndQuery()
         resultsJsonMultiLanguage.Records.First().Document.RootElement.GetProperty("name")
             .GetProperty("en-US").GetString().Should().Be("Azul");
     }
+
+    [TestMethod]
+    public async Task CreateCategoryInstance()
+    {
+        JsonSerializerOptions _serializerOptions = new JsonSerializerOptions();
```
Comment (ustims) [inline CloudFabric.EAV.Tests/JsonSerializationTests.cs:262]: I believe there is an options instance somewhere in tests which is passed to eav service. Is it possible to use that one?

**Review by ustims — CHANGES_REQUESTED** [review]

**Review by ustims — APPROVED** [review]

**Pull request merged**